### ISSUE TITLE
50 images were found but had errors...

### ISSUE THREAD

**Issue by gregbrunk** [boppreh/steamgrid#36]
First off - thank you - awesome program. Overlays make a huge different in my grid view. Well done.

What does the Unknown Format error mean? I would assume all the files pulled in from your DB would be in a compatible format. It found images for every single one of my side-loaded games, but couldn't apply to any of them.

**50 images were found but had errors and could not be overlaid:**
- Dynasty Warriors 8 Xtreme Legends (id 16100666639829172224) (image: unknown format)
- Battlefield 4 (id 13661906405827805184) (image: unknown format)
-  (id 223530) (image: unknown format)
- The Sims 3 (id 11589206328953798656) (image: unknown format)
- Dying Light (id 17207504682948231168) (image: unknown format)
- Morrowind (id 16499150497211482112) (image: unknown format)
- Need for Speed Rivals (id 10125426172478095360) (image: unknown format)
- TERA (id 14200117824365527040) (image: unknown format)
- Tom Clancy's Splinter Cell: Conviction (id 17217061723916206080) (image: unknown format)
- Diablo III (id 12273646364419686400) (image: unknown format)
- Assassin's Creed III (id 14780728417020018688) (image: unknown format)
- Far Cry 4 (id 13317026493373612032) (image: unknown format)
- Fallout New Vegas: Courier's Stash (id 22470) (image: unknown format)
- Hearthstone (id 17059105677311475712) (image: unknown format)
- Assassin's Creed IV (id 13836776600579866624) (image: unknown format)
- Dragon Age: Inquisition (id 16937699579057405952) (image: unknown format)
- Tom Clancy's Rainbox Six (id 10434579211229331456) (image: unknown format)
- Tom Clancy's Splinter Cell (id 17873429247545049088) (image: unknown format)
- Black & White 2 (id 17212212813213204480) (image: unknown format)
- Crysis 3 (id 18387128317061890048) (image: unknown format)
- Plants vs Zombies Garden Warfare (id 9948973960882290688) (image: unknown format)
- Tom Clancy's Splinter Cell: Chaos Theory (id 9498295234262466560) (image: unknown format)
-  (id 232210) (image: unknown format)
- The Sims 4 (id 12188511299340075008) (image: unknown format)
- Dead Space 3 (id 12804094754123939840) (image: unknown format)
- StarCraft (id 10006716974383497216) (image: unknown format)
- Guild Wars 2 (id 13873955164638937088) (image: unknown format)
- Tom Clancy's Rainbow Six 3: Raven Shield (id 14915648097295007744) (image: unknown format)
- Tom Clancy's Rainbow Six: Vegas 2 (id 11758388363508318208) (image: unknown format)
- SimCity (id 13991309751523213312) (image: unknown format)
-  (id 317170) (image: unknown format)
- Grand Theft Auto III (id 12230) (image: unknown format)
- Warcraft III (id 17148121421436682240) (image: unknown format)
- Tom Clancy's Ghost Recon: Future Soldier (id 10815022822988972032) (image: unknown format)
- Assassin's Creed Liberation H (id 13918776893584179200) (image: unknown format)
- Overwatch (id 13809453831118913536) (image: unknown format)
- Grand Theft Auto: San Andreas (id 12250) (image: unknown format)
- Dragon Age 2 (id 13402539133323182080) (image: unknown format)
- Oblivion (id 10661630557093363712) (image: unknown format)
- Tom Clancy's Rainbow Six: Vegas (id 16820529899105157120) (image: unknown format)
- Battlefield 3 (id 11491868175159525376) (image: unknown format)
- StarCraft II (id 13692624419950690304) (image: unknown format)
- Tom Clancy's Splinter Cell: Double Agent (id 14039418968695898112) (image: unknown format)
- World of Warcraft (id 10710089285819695104) (image: unknown format)
- Tom Clancy's EndWar Launcher (id 14111571941072568320) (image: unknown format)
- Grand Theft Auto: Vice City (id 12240) (image: unknown format)
- Mass Effect 3 (id 12283610735791046656) (image: unknown format)
- Tom Clancy's Ghost Recon (id 16211201175129686016) (image: unknown format)
- Titanfall 2 (id 12673584265800187904) (image: unknown format)
- Tom Clancy's Splinter Cell: Blacklist (id 12531381175689674752) (image: unknown format)

**Comment by gregbrunk** [comment]
Also, another slightly strange issue: Any manually added custom images (for side-loaded games) that I had added before using SteamGrid for the first time all received the correct overlay when the program was run. However, any custom images that I put in now, since running for the first time, are being ignored and overlays are not being applied.  Any ideas? I can make this a separate issue if you would like.

**Comment by boppreh** [comment]
> What does the Unknown Format error mean?

It downloaded the file and expected an image, but the file was something else. Currently jpeg and png are supported. I'll have to take a look what could be happening.

>  I would assume all the files pulled in from your DB would be in a compatible format. 

I don't have a database. The images are pulled either from the official Steam games, or by doing a google image search for the specific size.

> Also, another issue: Any manually added custom images (for side-loaded games) that I had added before using SteamGrid for the first time all received the correct overlay when SteamGrid was run. However, any custom images that I put in now, since running for the first time, are being ignored and overlays are not being applied when I re-run SteamGrid. Any ideas? I can make this a separate issue if you would like.

When it downloads an image, it also makes a backup copy. This is necessary in case you want to change the overlay, for example. It is probably pulling these backups instead of your new custom images. This is an issue that has no clear answer at the moment, but as a workaround you can go to the folder where the images are and manually erase the backup files for the games you want to customize the images.

I'll take a look at the other issues and give you a response later.

**Comment by gregbrunk** [comment]
> It downloaded the file and expected an image, but the file was something else. Currently jpeg and png are supported. I'll have to take a look what could be happening.

Interesting. It seemed like it was finding files, but wasn't actually downloading them because there was an error, because when I ran it in successive tries, it would say 50 images found but 0 downloaded. But I think thats because it did download them the first time, and now it is just checking them and they were all corrupt. 

> I don't have a database. The images are pulled either from the official Steam games, or by doing a google image search for the specific size.

Ah, my bad. I just assumed you were pulling from http://www.steamgriddb.com/. Not sure why I assumed your products were related.

> When it downloads an image, it also makes a backup copy. This is necessary in case you want to change the overlay, for example. It is probably pulling these backups instead of your new custom images. This is an issue that has no clear answer at the moment, but as a workaround you can go to the folder where the images are and manually erase the backup files for the games you want to customize the images.

So, could I solve this by going to `Steam/userdata/ID/config/grid` and deleting the files for my side-loaded games, manually adding custom images, then running the SteamGrid again? Or are the backups stored elsewhere?

**Comment by boppreh** [comment]
> So, could I solve this by going to Steam/userdata/ID/config/grid and deleting the files for my side-loaded games, manually adding custom images, then running the SteamGrid again? Or are the backups stored elsewhere?

Precisely. The code is at https://github.com/boppreh/steamgrid/blob/master/backup.go . The name of the file is the Steam ID of the game, and the backup has the same name but an "(original)" appended. If you delete both the image and its backup, then you can set a custom image again.

I know this should be simpler/clearer, but it's not clear to me how, other than just mentioning in the README. Any suggestions are welcome.

**Comment by gregbrunk** [comment]
Lovely! Thanks man. If you do have any ideas why it downloaded images for every one of my non-steam games, but was never able to use any of them, that would be great. I did look in the folder and every one of those files is a corrupted .jpg with a very long ID and wont open in windows, so could be related to https://github.com/boppreh/steamgrid/issues/32. I have attached one of the files if it helps.
![9948973960882290688 original](https://cloud.githubusercontent.com/assets/19507409/22269499/d3985878-e248-11e6-9d7e-209674c8cc0f.jpg)
